### PR TITLE
feat: add backend and frontend packages for iValt biometrics

### PR DESCRIPTION
**Pull request by KingOfTac**
# Pull Request

## Description

This PR adds two packages:
1. `@trans-stat/node-mfa` contains a simple backend wrapper around iValt's MFA API. The implementation is a simple javascript class that can be instantiated with an iValt API key. The implementation tries to be a simple as possible so that it is easy to extend and integrate with any node.js framework.
2. `@trans-stat/auth-components` is a library of FAST components for easy drop in user authentication UX patterns. Right now this library contains a single `<mfa-card>` component that is intended to be used with the node-mfa backend package. This library will be expanded overtime with more UX patterns related to user authentication.

### General

- [ ] I have added tests for my changes.
- [X] I have tested my changes.
- [X] I have updated the project documentation to reflect my changes.

## Next Steps

Next steps are:
1. ~~Create a MFA backend package for adding traditional 2FA and MFA capabilities to applications~~ This will be added to the node-mfa package
2. Add stackblitz examples demonstrating setting up the backend package with various backend frameworks
3. Add stackblitz examples demonstrating advanced usage of the frontend components and how to customize the Web Components.
4. ~~Add documentation on how to enable vscode intellisense for the Web Components in html and css files.~~ done

### CODE REVIEW

File: packages/web-components/src/biometric-card/biometric-card.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  public form!: HTMLFormElement;
+
+  /**
+   * @internal
+   */
+  public handleSubmit(): void {
+    const data = formValues(this.form) as BiometricCardFormData;
+
+    this.callback?.(data);
+    this.$emit('biometric-request', data);
```
Comment (scomea) [inline packages/web-components/src/biometric-card/biometric-card.ts]: What are your thoughts on event bubbling or not?  I've been tending to turn bubbling off as a micro-optimization in cases where I figure devs will tend to hook listeners directly to the component anyway as it is safe to turn it on later if there is a need.

**Review by scomea — APPROVED** [review]

File: packages/web-components/src/biometric-card/biometric-card.ts (hunk 1 above)
Comment (KingOfTac) [reply packages/web-components/src/biometric-card/biometric-card.ts]: What's the optimization benefit to turning it off? I'll admit I haven't dug too deep into configuring events beyond passing the name and event detail, so any help here would be appreciated.

**Comment by KingOfTac** [comment]
@bentomands added you as a reviewer to double check formatting and information in the documentation. Here's a list of the files to check:
- [README.md](https://github.com/trans-stat/tas-security/pull/10/files?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5)
- [packages/biometrics/README.md](https://github.com/trans-stat/tas-security/pull/10/files?short_path=ff32a39#diff-ff32a398c37ec90f9146c14790114fe6fd1befbb45abbb7aa4e2dc7ef1b3c7eb)
- [packages/web-components/README.md](https://github.com/trans-stat/tas-security/pull/10/files?short_path=ff32a39#diff-82b902c3378d024703bc5d540fd66b9f9c2e9b905d154f80029f46d2f0610040)

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,130 @@
+# Logs
```
Comment (awentzel) [inline .gitignore:1]: Can you remove all the ignore lines that are irrelevant to our technology stack, i.e.; `.grunt` for example?

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "version": "0.2.0",
+  "configurations": [
+    {
```
Comment (awentzel) [inline .vscode/launch.json]: what happens if user does not have edge?

File: README.md [hunk 4]
```diff
@@ -9,3 +9,6 @@ Protection provided by [QuSecure](https://www.qusecure.com)
 ### Multi-Factor Authentication
 Authentication provided by [iValt](https://www.ivalt.com)
 
+## Packages
+- [`@trans-stat/biometrics`](https://github.com/trans-stat/tas-security/blob/main/packages/biometrics) - Node.js esm and common.js compatible wrapper around iValt's biometric authentication API
```
Comment (awentzel) [inline README.md]: should `esm` be all caps?

File: beachball.config.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+module.exports = {
+	disallowedChangeTypes: ['major'],
+  access: 'public',
```
Comment (awentzel) [inline beachball.config.js:3]: formatting is off? Are we using prettier or something equivalent on merge to master things stay in desired format of the file type?

File: packages/web-components/README.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  "html.customData": [
+    "node_modules/@trans-stat/auth-components/dist/html-custom-data.json"
+  ],
+  "css.customData": [
+    "node_modules/@trans-stat/auth-components/dist/css-custom-data.json"
+  ]
+}
+```
+
+### `<biometric-card>`
```
Comment (awentzel) [inline packages/web-components/README.md]: iValt doesn't just provide biometrics, so I wonder if the naming is accurate here? For example, goefencing could be one factor of the MFA, which is not biometrics or even date time values.

File: packages/web-components/src/biometric-card/define.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+ * 
+ * @part card - The card element. The container for the form and the header and footer slots.
+ * @part form - The form element.
+ * @part mobile-number - The text field for capturing the user's mobile phone number.
+ * @part mobile-number__control - The mobile number field's internal control element.
+ * @part submit-button - The button used for submitting the form.
+ * @part submit-button__control - The submit button's internal control element.
+ */
+@customElement({
+  name: 'biometric-card',
```
Comment (awentzel) [inline packages/web-components/src/biometric-card/define.ts]: same question on naming? thoughts?

File: .gitignore (hunk 2 above)
Comment (KingOfTac) [reply .gitignore:1]: We could but it doesn't hurt anything. This is just the standard node.js gitignore that GitHub initializes repositories with.

File: packages/web-components/README.md (hunk 6 above)
Comment (KingOfTac) [reply packages/web-components/README.md]: I'm open to suggestions on naming. We could even just call this mfa-card and reuse it for other forms of multifactor.

File: beachball.config.js (hunk 5 above)
Comment (KingOfTac) [reply beachball.config.js:3]: Not at this time, no. Prettier is pretty heavy and very opinionated so I have been looking for alternatives that are smaller, faster, and easier to configure.

File: .vscode/launch.json (hunk 3 above)
Comment (KingOfTac) [reply .vscode/launch.json]: I'm actually removing this file since I am going a different route for local debugging and running.

**Comment by KingOfTac** [comment]
@awentzel I renamed the backend package to be more generic so that other MFA service and API wrappers can live there in addition to the iValt wrapper. I also renamed the frontend component to be more generic and am planning on more changes to the component's API in the near future that will enable it to be used for a wider variety of scenarios.

I also added a wildcard export to the backend package so that specific wrappers can be imported and the rest tree shaken out if needed (note this is only if the application is using ESM since the CommonJS version is bundled only)

File: packages/web-components/src/biometric-card/biometric-card.ts (hunk 1 above)
Comment (scomea) [reply packages/web-components/src/biometric-card/biometric-card.ts]: I assume that there is more work going on in the background to handle a bubbling event vs. a non-bubbling, but I haven't tested to see what impact it actually has.  To control bubbling you pass it as an option `this.$emit("someEvent", eventDetail, {bubbles: false});`

**Review by awentzel — APPROVED** [review]

**Pull request merged**